core/services/gateway/network/httpserver_test.go

```diff
@@ -0,0 +1,71 @@
+package network_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/network/mocks"
+)
+
+func startNewServer(t *testing.T, maxRequestBytes int64, readTimeoutMillis uint32) (server network.HttpServer, handler *mocks.HTTPRequestHandler, url string) {
+	config := &network.HTTPServerConfig{
+		Host:                 "localhost",
+		Port:                 0,
+		Path:                 "/test_path",
+		TLSEnabled:           false,
+		ContentTypeHeader:    "application/jsonrpc",
+		ReadTimeoutMillis:    readTimeoutMillis,
+		WriteTimeoutMillis:   10_000,
+		RequestTimeoutMillis: 10_000,
+		MaxRequestBytes:      maxRequestBytes,
+	}
+
+	handler = mocks.NewHTTPRequestHandler(t)
+	server = network.NewHttpServer(config, logger.TestLogger(t))
+	server.SetHTTPRequestHandler(handler)
+	err := server.Start(testutils.Context(t))
+	require.NoError(t, err)
+
+	port := server.GetPort()
+	url = fmt.Sprintf("http://localhost:%d/test_path", port)
```

Comment:
Now you have `localhost` hardcoded in two places. It would be better instead of `GetPort()` to add `GetAddr()` that would return complete address:port string.